### ISSUE TITLE
Incorrect exception thrown

### ISSUE THREAD

**Issue by glensc** [jawah/niquests#162]
low level exceptions should be rethrown with niquests exceptions.

This allows application to take appropriate action, handler for TimeoutException or ServerError. or catch any RequestException.

## Actual Result

```
         │ /lib/python3.9/site-packages/urllib3/backend/hface.py:852 in       │
         │ __exchange_until                                                                                                                  │
         │                                                                                                                                   │
         │    849 │   │   │   │   │   │                                                                                                      │
         │    850 │   │   │   │   │   │   self.sock.sendall(data_out)                                                                        │
         │    851 │   │   │   │                                                                                                              │
         │ ❱  852 │   │   │   │   data_in = self.sock.recv(self.blocksize)                                                                   │
         │    853 │   │   │   │                                                                                                              │
         │    854 │   │   │   │   if not data_in:                                                                                            │
         │    855 │   │   │   │   │   # in some cases (merely http/1 legacy)                                                                 │
         timeout: timed out                                                                                                                   
```

## System Information

    $ python -m niquests.help

<details>

```json
{
  "charset_normalizer": {
    "version": "3.4.0"
  },
  "http1": {
    "h11": "0.14.0"
  },
  "http2": {
    "jh2": "5.0.3"
  },
  "http3": {
    "enabled": true,
    "qh3": "1.2.1"
  },
  "idna": {
    "version": "3.10"
  },
  "implementation": {
    "name": "CPython",
    "version": "3.9.20"
  },
  "niquests": {
    "version": "3.10.0"
  },
  "ocsp": {
    "enabled": true
  },
  "platform": {
    "release": "23.6.0",
    "system": "Darwin"
  },
  "system_ssl": {
    "version": "30300020"
  },
  "urllib3.future": {
    "cohabitation_version": null,
    "version": "2.11.901"
  },
  "wassima": {
    "certifi_fallback": false,
    "enabled": true,
    "version": "1.1.4"
  }
}

```

</details>

**Comment by Ousret** [comment]
I'll need the complete stacktrace to address this.

**Comment by glensc** [comment]
```
2024-10-22 01:26:26,336 DEBUG[trakt.core]:method, url :: get, https://api.trakt.tv/search/tmdb/560144?type=movie
2024-10-22 01:26:27,326 DEBUG[quic]:[91694d0e0c9ebbe8] Remote max_streams_bidi raised to 427
2024-10-22 01:28:27,343 ERROR[plextraktsync.cli]:timed out
Traceback (most recent call last):
  File "plextraktsync/cli.py", line 28, in wrap
    cmd(*args, **kwargs)
  File "plextraktsync/commands/sync.py", line 74, in sync
    run_async(runner, walker=w, dry_run=config.dry_run)
  File "python-3.9/lib/python3.9/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "plextraktsync/decorators/coro.py", line 15, in coro
    return asyncio.run(f(*args, **kwargs))
  File "/opt/homebrew/Cellar/python@3.9/3.9.20/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.9/3.9.20/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "plextraktsync/commands/sync.py", line 13, in run_async
    await runner.sync(**kwargs)
  File "plextraktsync/sync/Sync.py", line 49, in sync
    async for movie in walker.find_movies():
  File "plextraktsync/plan/Walker.py", line 98, in find_movies
    movie = self.mf.resolve_any(plex)
  File "plextraktsync/media/MediaFactory.py", line 39, in resolve_any
    m = self.resolve_guid(guid, show)
  File "plextraktsync/media/MediaFactory.py", line 68, in resolve_guid
    tm = self.trakt.find_by_guid(guid)
  File "plextraktsync/trakt/TraktApi.py", line 259, in find_by_guid
    tm = self.search_by_id(guid.id, id_type=guid.provider, media_type=guid.type)
  File "python-3.9/lib/python3.9/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "plextraktsync/decorators/rate_limit.py", line 20, in rate_limit
    return fn(*args, **kwargs)
  File "python-3.9/lib/python3.9/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "plextraktsync/decorators/retry.py", line 26, in retry
    return fn(*args, **kwargs)
  File "plextraktsync/trakt/TraktApi.py", line 306, in search_by_id
    search = trakt.sync.search_by_id(media_id, id_type=id_type, media_type=media_type)
  File "python-3.9/lib/python3.9/site-packages/trakt/core.py", line 579, in inner
    json_data = self._handle_request('get', url)
  File "python-3.9/lib/python3.9/site-packages/trakt/core.py", line 544, in _handle_request
    response = session.request(method, url, headers=HEADERS,
  File "python-3.9/lib/python3.9/site-packages/requests_cache/session.py", line 183, in request
    return super().request(method, url, *args, headers=headers, **kwargs)  # type: ignore
  File "python-3.9/lib/python3.9/site-packages/niquests/sessions.py", line 563, in request
    resp = self.send(prep, **send_kwargs)
  File "python-3.9/lib/python3.9/site-packages/requests_cache/session.py", line 230, in send
    response = self._send_and_cache(request, actions, cached_response, **kwargs)
  File "python-3.9/lib/python3.9/site-packages/requests_cache/session.py", line 255, in _send_and_cache
    actions.update_from_response(response)
  File "python-3.9/lib/python3.9/site-packages/requests_cache/policy/actions.py", line 224, in update_from_response
    directives = CacheDirectives.from_headers(response.headers)
  File "python-3.9/lib/python3.9/site-packages/niquests/models.py", line 1093, in __getattribute__
    super().__getattribute__("_gather")()
  File "python-3.9/lib/python3.9/site-packages/niquests/models.py", line 1073, in _gather
    super().__getattribute__("connection").gather(self)
  File "python-3.9/lib/python3.9/site-packages/niquests/adapters.py", line 1276, in gather
    low_resp = self.poolmanager.get_response(
  File "python-3.9/lib/python3.9/site-packages/urllib3/poolmanager.py", line 570, in get_response
    response = pool.get_response(promise=promise)
  File "python-3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 815, in get_response
    response = conn.getresponse(promise=promise, police_officer=self.pool)
  File "python-3.9/lib/python3.9/site-packages/urllib3/connection.py", line 596, in getresponse
    low_response = super().getresponse(
  File "python-3.9/lib/python3.9/site-packages/urllib3/backend/hface.py", line 1364, in getresponse
    head_event: HeadersReceived | EarlyHeadersReceived = self.__exchange_until(  # type: ignore[assignment]
  File "python-3.9/lib/python3.9/site-packages/urllib3/backend/hface.py", line 852, in __exchange_until
    data_in = self.sock.recv(self.blocksize)
socket.timeout: timed out
```

**Comment by glensc** [comment]
one more, with recent urllib3.future==2.11.902. trace is same, just lines are different

```
2024-10-22 12:39:23,151 DEBUG[trakt.core]:method, url :: get, https://api.trakt.tv/search/tmdb/44943?type=movie
2024-10-22 12:39:23,209 DEBUG[quic]:[75466843281896b7] Remote max_streams_bidi raised to 531
2024-10-22 12:41:23,228 ERROR[plextraktsync.cli]:timed out
Traceback (most recent call last):
  File "plextraktsync/cli.py", line 28, in wrap
    cmd(*args, **kwargs)
  File "plextraktsync/commands/sync.py", line 74, in sync
    run_async(runner, walker=w, dry_run=config.dry_run)
  File "python-3.9/lib/python3.9/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "plextraktsync/decorators/coro.py", line 15, in coro
    return asyncio.run(f(*args, **kwargs))
  File "/opt/homebrew/Cellar/python@3.9/3.9.20/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.9/3.9.20/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "plextraktsync/commands/sync.py", line 13, in run_async
    await runner.sync(**kwargs)
  File "plextraktsync/sync/Sync.py", line 49, in sync
    async for movie in walker.find_movies():
  File "plextraktsync/plan/Walker.py", line 98, in find_movies
    movie = self.mf.resolve_any(plex)
  File "plextraktsync/media/MediaFactory.py", line 39, in resolve_any
    m = self.resolve_guid(guid, show)
  File "plextraktsync/media/MediaFactory.py", line 68, in resolve_guid
    tm = self.trakt.find_by_guid(guid)
  File "plextraktsync/trakt/TraktApi.py", line 259, in find_by_guid
    tm = self.search_by_id(guid.id, id_type=guid.provider, media_type=guid.type)
  File "python-3.9/lib/python3.9/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "plextraktsync/decorators/rate_limit.py", line 20, in rate_limit
    return fn(*args, **kwargs)
  File "python-3.9/lib/python3.9/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "plextraktsync/decorators/retry.py", line 26, in retry
    return fn(*args, **kwargs)
  File "plextraktsync/trakt/TraktApi.py", line 306, in search_by_id
    search = trakt.sync.search_by_id(media_id, id_type=id_type, media_type=media_type)
  File "python-3.9/lib/python3.9/site-packages/trakt/core.py", line 579, in inner
    json_data = self._handle_request('get', url)
  File "python-3.9/lib/python3.9/site-packages/trakt/core.py", line 544, in _handle_request
    response = session.request(method, url, headers=HEADERS,
  File "python-3.9/lib/python3.9/site-packages/requests_cache/session.py", line 183, in request
    return super().request(method, url, *args, headers=headers, **kwargs)  # type: ignore
  File "python-3.9/lib/python3.9/site-packages/niquests/sessions.py", line 563, in request
    resp = self.send(prep, **send_kwargs)
  File "python-3.9/lib/python3.9/site-packages/requests_cache/session.py", line 230, in send
    response = self._send_and_cache(request, actions, cached_response, **kwargs)
  File "python-3.9/lib/python3.9/site-packages/requests_cache/session.py", line 255, in _send_and_cache
    actions.update_from_response(response)
  File "python-3.9/lib/python3.9/site-packages/requests_cache/policy/actions.py", line 224, in update_from_response
    directives = CacheDirectives.from_headers(response.headers)
  File "python-3.9/lib/python3.9/site-packages/niquests/models.py", line 1093, in __getattribute__
    super().__getattribute__("_gather")()
  File "python-3.9/lib/python3.9/site-packages/niquests/models.py", line 1073, in _gather
    super().__getattribute__("connection").gather(self)
  File "python-3.9/lib/python3.9/site-packages/niquests/adapters.py", line 1276, in gather
    low_resp = self.poolmanager.get_response(
  File "python-3.9/lib/python3.9/site-packages/urllib3/poolmanager.py", line 570, in get_response
    response = pool.get_response(promise=promise)
  File "python-3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 815, in get_response
    response = conn.getresponse(promise=promise, police_officer=self.pool)
  File "python-3.9/lib/python3.9/site-packages/urllib3/connection.py", line 596, in getresponse
    low_response = super().getresponse(
  File "python-3.9/lib/python3.9/site-packages/urllib3/backend/hface.py", line 1394, in getresponse
    head_event: HeadersReceived | EarlyHeadersReceived = self.__exchange_until(  # type: ignore[assignment]
  File "python-3.9/lib/python3.9/site-packages/urllib3/backend/hface.py", line 864, in __exchange_until
    data_in = self.sock.recv(self.blocksize)
socket.timeout: timed out
```

**Comment by Ousret** [comment]
It's fixed. See linked PR.
Published in v3.10.1

regards,
